### DISCUSSION TITLE
Error when building the project at JKube Kit :: Build :: API

### DISCUSSION THREAD

**Discussion by varad64** [eclipse/jkube#2023] [Q&A]
Hello, I am new to the project and tried to build the project but got the following error on Windows 10.

```
[ERROR] Tests run: 9, Failures: 8, Errors: 0, Skipped: 0, Time elapsed: 1.688 s <<< FAILURE! - in org.eclipse.jkube.kit.build.api.assembly.AssemblyManagerCreateDockerTarArchiveTest
[ERROR] org.eclipse.jkube.kit.build.api.assembly.AssemblyManagerCreateDockerTarArchiveTest.withoutDockerfileAndAlreadyExistingFileInAssemblyGetsOverwritten  Time elapsed: 1.031 s  <<< FAILURE!
java.lang.AssertionError:

Expecting ArrayList:
  ["Dockerfile",
    "jkube-generated-layer-final-artifact",
    "jkube-generated-layer-final-artifact\maven",
    "jkube-generated-layer-final-artifact\maven\test-0.1.0.jar",
    "maven"]
to contain only:
  ["Dockerfile",
    "jkube-generated-layer-final-artifact",
    "jkube-generated-layer-final-artifact/maven",
    "jkube-generated-layer-final-artifact/maven/test-0.1.0.jar",
    "maven"]
element(s) not found:
  ["jkube-generated-layer-final-artifact/maven",
    "jkube-generated-layer-final-artifact/maven/test-0.1.0.jar"]
and element(s) not expected:
  ["jkube-generated-layer-final-artifact\maven",
    "jkube-generated-layer-final-artifact\maven\test-0.1.0.jar"]

        at org.eclipse.jkube.kit.build.api.assembly.AssemblyManagerCreateDockerTarArchiveTest.withoutDockerfileAndAlreadyExistingFileInAssemblyGetsOverwritten(AssemblyManagerCreateDockerTarArchiveTest.java:164)
```

Has this issue got to do anything with the slashes in the path. Very confused. Any help would be appreciated.

Here's the complete surefire report file
[org.eclipse.jkube.kit.build.api.assembly.AssemblyManagerCreateDockerTarArchiveTest.txt](https://github.com/eclipse/jkube/files/10507724/org.eclipse.jkube.kit.build.api.assembly.AssemblyManagerCreateDockerTarArchiveTest.txt)

Thanks in advance!

**Comment by manusa** [comment]
Yes it does.
There was some work in progress to improve the tests and make them Windows-compatible.
If you want to build the project in Windows you'll need to skip the tests:
```
mvn package -DskipTests
```
Or if you want to install to your local Maven repository
```
mvn install -DskipTests
```

**Reply by varad64** [reply]
Oh alright, I will try this out. Thanks a lot!

**Comment by varad64** [comment]
Hello @manusa, the above command skipped the tests but there's a different error now, which is related to the maven-javadoc-plugin. 

The error is as below:

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.4.1:jar (default) on project jkube-kit-build-service-docker: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - E:\jkube\jkube-kit\build\service\docker\target\generated-sources\delombok\org\eclipse\jkube\kit\build\service\docker\DockerAccessFactory.java:145: warning: no @param for projectProperties
[ERROR]             public DockerAccessFactory.DockerAccessContext.DockerAccessContextBuilder projectProperties(final Properties projectProperties) {

Similar error where @param is missing in multiple places for multiple files. I tried adding in the 
`<additionalparam>-Xdoclint:none</additionalparam>` parameter to have the plugin skip the checking of javadoc but still am presented with the same error.

Should I skip javadoc generation altogether or is there some other alternative?

Thanks!

**Reply by manusa** [reply]
I'm not sure why this goal is executing for you and why the warnings make your build fail. Are you running maven from the command line?

**Reply by varad64** [reply]
I'm using git bash

**Reply by manusa** [reply]
just use your regular windows `CMD.exe` instead

**Reply by varad64** [reply]
Okay, I'll do that and let you know how it goes. Thanks

**Reply by varad64** [reply]
Still getting the same error on cmd as well

**Reply by manusa** [reply]
I'll try later today, maybe something changed.

**Reply by varad64** [reply]
Okay, thanks a lot for your time!

**Reply by varad64** [reply]
Update: Tried building the project on a Linux machine using a VM and it builds successfully. Could be something to do with how the project is being set up on Windows.
